xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenOrderBookUtils.java

```diff
@@ -97,4 +92,26 @@ private static long timestampToMs(String timestamp) {
         return new BigDecimal(timestamp).multiply(new BigDecimal(1000)).longValue();
     }
 
+    public static OrderBook verifyKrakenOrderBook(OrderBook orderBook) {
+
+        Collections.sort(orderBook.getBids());
```

Comment:
The sort looks wrong. Order books are usually sorted in descending order on one side and ascending on the other.

Reply:
Will this : Collections.sort(orderBook.getBids(),Collections.reverseOrder()); fix the issue with sorting?

Reply:
@makarid 
Always order book items are sorted by price: asks with ascending order and bids with descending order. See TreeSets here:  info/bitrich/xchangestream/kraken/KrakenOrderBookStorage.java:40
So you don't need to sort items before processing. 

If you have sorted bids in ascending order then order book became invalid in bids. 
So there is the question here did u test you changes before pool request created :)